contrib/nydus-snapshotter/pkg/store/cachestore.go

```diff
@@ -22,23 +29,64 @@ func NewCacheStore(db *Database) (*CacheStore, error) {
 }
 
 func (cs *CacheStore) AddSnapshot(imageID string, blobs []string) error {
+	cs.Lock()
+	defer cs.Unlock()
+
 	ss := &Snapshot{
 		ImageID:  imageID,
 		Blobs:    blobs,
 		CreateAt: time.Now(),
 		UpdateAt: time.Now(),
 	}
-	if err := cs.Database.AddSnapshot(imageID, ss); err != nil {
+	if err := cs.Database.addSnapshot(imageID, ss); err != nil {
 		return err
 	}
 	for _, id := range blobs {
 		blob := &Blob{
 			CreateAt: time.Now(),
 			UpdateAt: time.Now(),
 		}
-		if err := cs.Database.AddBlob(id, blob); err != nil {
+		if err := cs.Database.addBlob(id, blob); err != nil {
 			return err
 		}
 	}
 	return nil
 }
+
+func (cs *CacheStore) DelSnapshot(imageID string) error {
+	cs.Lock()
+	defer cs.Unlock()
+
+	return cs.Database.delSnapshot(imageID)
+
+}
+
+func (cs *CacheStore) GC(delFunc func(blob string) error) ([]string, error) {
+	cs.Lock()
+	defer cs.Unlock()
```

Comment:
Hopefully, we don't have too many blobs on the same host. So it's fine to have a big lock here and we should optimize it when we see stalls with blob GCs.